pints/interfaces/_stan.py

```diff
@@ -0,0 +1,101 @@
+#
+# Interface for Stan models
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+from collections import Counter
+import pystan
+
+from . import InterfaceLogPDF
+
+
+class StanLogPDF(InterfaceLogPDF):
+    def __init__(self, stan_code, stan_data):
+
+        print("Compiling Stan model then test running...")
+        sm = pystan.StanModel(model_code=stan_code)
+        stanfit = sm.sampling(data=stan_data, iter=1, chains=1,
+                              verbose=False, refresh=10,
+                              control={'adapt_engaged': False})
+        print("Stan model compiled and runs ok...ignore various warnings.")
+        self._compiled_stan = sm
+        self._fit = stanfit
+        self._log_prob = stanfit.log_prob
+        self._grad_log_prob = stanfit.grad_log_prob
+        self._u_to_c = stanfit.unconstrain_pars
+        names = stanfit.unconstrained_param_names()
+        self._n_parameters = len(names)
+        self._names, self._index = self._initialise_dict_index(names)
+        self._long_names = names
+        self._counter = Counter(self._index)
+        self._dict = {self._names[i]: [] for i in range(len(self._names))}
+
+    def __call__(self, x):
+        dict = self._dict_update(x)
+        try:
+            return self._log_prob(self._u_to_c(dict), adjust_transform=True)
+        # if Pints proposes a value outside of Stan's parameter bounds
```

Comment:
Some way to check this? Or some more specific error we could catch?
RuntimeError can be a lot of things

Reply:
yea, this caught me out recently! Need to catch the specific error